src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml

```diff
@@ -0,0 +1,148 @@
+            </Button>
+            <Button Grid.Column="3" Click="SortByOrder" ToolTip="{x:Static self:DocumentOutlineStrings.Document_Outline_Sort_Order}">
+                <imaging:CrispImage Moniker="{x:Static imagecatalog:KnownMonikers.SortByNamespace}" />
+            </Button>
+            <Button Grid.Column="4" Click="SortByType" ToolTip="{x:Static self:DocumentOutlineStrings.Document_Outline_Sort_Type}">
+                <imaging:CrispImage Moniker="{x:Static imagecatalog:KnownMonikers.SortByType}" />
+            </Button>
+        </Grid>
+        <TextBox Grid.Row="1" x:Name="searchBox" TextChanged="Search" VerticalContentAlignment="Center">
+            <TextBox.Style>
+                <Style TargetType="TextBox">
+                    <Setter Property="Background" Value="{StaticResource {x:Static vsshell:VsBrushes.ToolWindowBackgroundKey}}" />
+                    <Setter Property="Foreground" Value="{DynamicResource {x:Static vsshell:VsBrushes.ToolWindowTextKey}}" />
+                    <Setter Property="Padding" Value="3, 0, 0, 0" />
+                    <Setter Property="FontFamily" Value="{StaticResource {x:Static vsshell:VsFonts.CaptionFontFamilyKey}}" />
+                    <Setter Property="FontSize" Value="{StaticResource {x:Static vsshell:VsFonts.CaptionFontSizeKey}}" />
+                    <Setter Property="CaretBrush" Value="{StaticResource {x:Static vsui:HeaderColors.DefaultTextBrushKey}}" />
+                    <Style.Resources>
+                        <VisualBrush x:Key="CueBannerBrush" AlignmentX="Left" AlignmentY="Center" Stretch="None">
+                            <VisualBrush.Visual>
+                                <Label Content="{x:Static self:DocumentOutlineStrings.Document_Outline_Search}"
+                                       Foreground="{DynamicResource {x:Static vsshell:VsBrushes.TitleBarInactiveTextKey}}"
+                                       FontFamily="{StaticResource {x:Static vsshell:VsFonts.CaptionFontFamilyKey}}"
+                                       FontSize="{StaticResource {x:Static vsshell:VsFonts.CaptionFontSizeKey}}" />
+                            </VisualBrush.Visual>
+                        </VisualBrush>
+                    </Style.Resources>
+                    <Style.Triggers>
+                        <Trigger Property="Text" Value="">
+                            <Setter Property="Background" Value="{StaticResource CueBannerBrush}" />
+                        </Trigger>
+                    </Style.Triggers>
+                </Style>
+            </TextBox.Style>
+        </TextBox>
+        <TreeView Grid.Row="2" Name="symbolTree">
+            <TreeView.Resources>
+                <HierarchicalDataTemplate DataType="{x:Type self:DocumentSymbolItem}" ItemsSource="{Binding Children}">
+                    <StackPanel Orientation="Horizontal" VerticalAlignment="Center" MouseDown="JumpToContent">
```

Comment:
nit/question for xaml experts.  is MouseDown generally what is listened for?  Or is there some event that indicates more specifically 'this was selected'?

Reply:
You can subscribe instead to [`SelectionChanged`](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.primitives.selector.selectionchanged?view=netframework-4.8#remarks) or [`GotFocus`](https://docs.microsoft.com/en-us/dotnet/api/system.windows.uielement.gotfocus?view=netframework-4.8) which will call you anytime the selection is modified (even programmatically). But neither of these methods allow you to differentiate between user input and us changing the item selected based on the cursor position in the text editor.

[`MouseDown`](https://docs.microsoft.com/en-us/dotnet/api/system.windows.uielement.mousedown?view=netframework-4.8), [`KeyDown`](https://docs.microsoft.com/en-us/dotnet/api/system.windows.uielement.keydown?view=netframework-4.8), [`TouchDown`](https://docs.microsoft.com/en-us/dotnet/api/system.windows.uielement.touchdown?view=netframework-4.8) and [`StylusDown`](https://docs.microsoft.com/en-us/dotnet/api/system.windows.uielement.stylusdown?view=netframework-4.8) guarantee that you are responding to user input.

@emilyanas2323 we may want to respond to `KeyDown` events in addition to `MouseDown` events so keyboard selection of items in the tree works. I am unfamiliar with whether stylus or touch inputs are expected to work in VS, @ryzngard do you know?